browser/misc_metrics/sources.gni

```diff
@@ -8,8 +8,11 @@ brave_browser_misc_metrics_sources = [
   "//brave/browser/misc_metrics/profile_misc_metrics_service.cc",
   "//brave/browser/misc_metrics/profile_misc_metrics_service_factory.cc",
   "//brave/browser/misc_metrics/uptime_monitor.cc",
-  "//brave/browser/misc_metrics/usage_clock.cc",
 ]
+if (!is_android) {
+  brave_browser_misc_metrics_sources +=
+      [ "//brave/browser/misc_metrics/usage_clock.cc" ]
```

Comment:
why is this being added directly to sources.gni?

Reply:
this combination of BUILD.gn and sources.gni here is not good in general and needs to be addressed, but for now at least this file should move into the BUILD.gn

Reply:
I guess in this case there are chrome/browser deps for 
```
base::TimeTicks NowTicks() {
  return g_tick_clock_for_testing ? g_tick_clock_for_testing->NowTicks()
                                  : base::TimeTicks::Now();
}
```
but that doesn't seem necessary, can't time already be overridden in testing without this?

Reply:
what are we using this for anyway?

Reply:
cc: @DJAndries

Reply:
> what are we using this for anyway? Do we really need this metric?

We are using the usage clock for this metric: https://github.com/brave/brave-browser/wiki/P3A#how-long-has-the-browser-been-active-for-in-the-past-day

It's a pretty critical one.

Reply:
@bridiver the dependency on `//chrome/browser` is: 
```
ERROR at //brave/browser/misc_metrics/usage_clock.cc:8:11: Include not allowed.
#include "chrome/browser/resource_coordinator/time.h"
          ^-----------------------------------------
The include file is in the target(s):
  //chrome/browser:browser
```

Reply:
that resource_coordinator time class looks like it only exists to mock the time which we can do without that class so why don't we just drop it?

Reply:
but also adding to sources.gni is not the right way to fix this. You should use `brave_chrome_browser_allow_circular_includes_from`

Reply:
Updated in https://github.com/brave/brave-core/pull/27522/commits/7581e216fa3bd56afe489643b52369d4a30b4092.